src/aatoolbox/datasources/iri/iri_forecast_seasonal_precipitation_tercile.py

```diff
@@ -0,0 +1,184 @@
+            iri key for authentication. An account is
+            needed to get this key config. For an account this key might
+            be changed over time, so might need to update it regularly
+        dominant: bool
+            Two datasets with the tercile forecast exists. One reports
+            the probability per tercile, which is downloaded with
+            dominant=False. The other only reports the probability
+            of the dominant tercile, which is downloaded with
+            dominant=True. In this data low negative probabilities indicate the
+            below average tercile is dominant and high positive that the above
+            average tercile is dominant.
+
+        Examples
+        --------
+        >>> from aatoolbox.pipeline import Pipeline
+        >>> from aatoolbox.utils.geoboundingbox import GeoBoundingBox
+        >>> (from aatoolbox.datasources.iri.
+        ... iri_forecast_seasonal_precipitation_tercile import IriForecast)
+        >>> iso3="bfa"
+        >>> pipeline_iso = Pipeline(iso3)
+        >>> codab_admin1 = pipeline_iso.load_codab(admin_level=1)
+        >>> geo_bounding_box = GeoBoundingBox.from_shape(codab_admin1)
+        >>> iri=IriForecast(iso3,geo_bounding_box)
+        >>> iri.download(os.getenv("IRI_AUTH"))
+        """
+        # question: I chose to always remove the file and redownload
+        # for all dates instead of allowing a selection of dates.
+        # This because the file is <1MB and only takes about
+        # a minute to download. Do you think this makes sense?
+
+        output_filepath = self._get_raw_path(dominant=dominant)
+        # strange things happen when just overwriting the file, so delete it
+        # first if it already exists
+        output_filepath.unlink(missing_ok=True)
+        output_filepath.parent.mkdir(parents=True, exist_ok=True)
+
+        url = self._get_url(dominant=dominant)
+        logger.info("Downloading IRI NetCDF file.")
+        response = requests.get(
+            url,
+            # have to authenticate by using a cookie
+            cookies={"__dlauth_id": iri_auth},
+        )
+        with open(output_filepath, "wb") as out_file:
+            out_file.write(response.content)
+
+        return output_filepath
+
+    def _get_url(self, dominant: bool):
+        # question: best to define url here or as global constant?
+        base_url = (
+            "https://iridl.ldeo.columbia.edu/SOURCES/.IRI/.FD/"
+            ".NMME_Seasonal_Forecast/.Precipitation_ELR/"
+        )
+        if dominant:
+            base_url += ".dominant/"
+        else:
+            base_url += ".prob/"
+        return (
+            f"{base_url}"
+            f"X/%28{self._geobb.west}%29%28{self._geobb.east}%29RANGEEDGES/"
+            f"Y/%28{self._geobb.north}%29%28{self._geobb.south}%29RANGEEDGES/"
+            "data.nc"
+        )
+
+    def load(self, dominant: bool = False):
+        """Preprocess and load the IRI forecast.
```

Comment:
You should separate the processing and load steps

Reply:
Okay! But I don't want to save a separate processed file since the processing is so fast. So does it then make sense to you to just make the process function a private function? (`def _process(ds)`)

Reply:
I would say it's more about having a consistent architecture for every datasource, i.e. having a `download`, `process`, and `load` method, regardless of the speed.